src/main/java/com/arpnetworking/metrics/impl/TsdMetricsFactory.java

```diff
@@ -107,37 +205,66 @@ protected TsdMetricsFactory(final Builder builder) {
     public static class Builder {
 
         /**
+         * Public constructor.
+         */
+        public Builder() {
```

Comment:
What you describe is a common implementation of the Builder pattern. However, I have some issues with this. First, it does not work well with most deserialization libraries which require an empty constructor (yes, I realize that does not apply directly here, but I would prefer a consistent pattern across the codebase). Second, it leaves open the large constructor problem; consider a class with N required fields, then you have a builder constructor with N arguments and what no setters? Happy to discuss further.